src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftPluginDescriptor.java

```diff
@@ -61,13 +61,13 @@ default ListBoxModel doFillWaitUnitItems() {
     default FormValidation doTestConnection(@QueryParameter String apiURL, @QueryParameter String authToken) {
 
 
-        if (StringUtils.isEmpty(apiURL)) {
-            if(!EnvVars.masterEnvVars.containsKey(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY) &&
-                    !StringUtils.isEmpty(EnvVars.masterEnvVars.get(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY))) {
+        if (apiURL == null || StringUtils.isEmpty(apiURL)) {
+            if(!EnvVars.masterEnvVars.containsKey("KUBERNETES_MASTER") &&
+                    !StringUtils.isEmpty(EnvVars.masterEnvVars.get("KUBERNETES_MASTER"))) {
```

Comment:
Also, I'm wondering what you see when comparing `EnvVars.masterEnvVars` with `Computer.currentComputer().getEnvironment()` - perhaps my original suggestion to you was not ideal.

The typical job execution path seems to pull env vars from a number of sources.  Computer is one of them.